src/runtime/ndarray.cc

```diff
@@ -68,8 +68,16 @@ void NDArray::Internal::DefaultDeleter(NDArray::Container* ptr) {
   } else if (ptr->dl_tensor.data != nullptr) {
     // if the array is still pinned before freeing, unpin it.
     if (ptr->pinned_by_dgl_) UnpinContainer(ptr);
-    dgl::runtime::DeviceAPI::Get(ptr->dl_tensor.ctx)
-        ->FreeDataSpace(ptr->dl_tensor.ctx, ptr->dl_tensor.data);
+    if (ptr->pinned_by_pyt_) {
+      DeviceAPI::Get(kDGLCUDA)->FreePinnedDataSpace(&(ptr->pyt_raw_deleter));
+      CHECK(ptr->pyt_raw_deleter == nullptr);
```

Comment:
Just to make sure: this is an assertion, right?

Reply:
Yes, it is an assertion, as I assign `pyt_raw_deleter` to nullptr when the deleter is called (see `torch.cpp` L85).